checksum/src/test/scala/uk/gov/nationalarchives/checksum/ChecksumGeneratorTest.scala

```diff
@@ -0,0 +1,41 @@
+package uk.gov.nationalarchives.checksum
+
+import org.mockito.{ArgumentCaptor, MockitoSugar}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+import org.scalatest.prop.TableDrivenPropertyChecks
+
+import scala.jdk.CollectionConverters._
+
+class ChecksumGeneratorTest extends AnyFlatSpec with MockitoSugar with TableDrivenPropertyChecks {
+
+  val fileSizes =
+    Table(
+      "fileSizeInMb",
+      1,
+      10,
+      200
+    )
+
+  forAll(fileSizes) { fileSizeInMb => {
+    "The generate method" should s"calls the bucket client with the correct start and end values for file size $fileSizeInMb" in {
```

Comment:
Should be: "... call the bucket client ..."

Reply:
Good spot, I've changed it.